rnaseq-cgl-pipeline/Dockerfile-template.py

```diff
@@ -0,0 +1,117 @@
+import argparse
+import os
+import textwrap
+
+import subprocess
+
+# Using Docker versions from: https://github.com/docker/docker/blob/master/CHANGELOG.md
+docker_versions = ['1.7.0',
+                   '1.7.1',
+                   '1.8.0',
+                   '1.8.1',
+                   '1.9.0',
+                   '1.9.1',
+                   '1.10.0',
+                   '1.10.1',
+                   '1.10.3']
+
+
+def build(name, tag, version):
+    """
+    Builds a container given a name, tag, and version of Docker
+
+    :param str name: Name of the image
+    :param str tag: Docker tag, consisting of git commit hashes
+    :param str version: Docker version
+    """
+    dirname, filename = os.path.split(os.path.abspath(__file__))
+    version_tag = version + '--' + tag
+    # Write out Dockerfile
+    with open(os.path.join(dirname, 'Dockerfile'), 'w') as f:
+        f.write(print_dockerfile(version))
+    # Build
+    subprocess.check_call(['docker', 'build',
+                           '-t', '{}:{}'.format(name, version_tag),
+                           dirname])
+    # Tag
+    subprocess.check_call(['docker', 'tag',
+                           '-f', '{}:{}'.format(name, version_tag),
+                           '{}:latest'.format(name)])
+    os.remove(os.path.join(dirname, 'Dockerfile'))
```

Comment:
I would not remove after building. Add a .gitignore entry for the Dockerfiles. If you do feel strongly about removing the dockerfiles, at least make it easy to override.